templates/single-product/related.php

```diff
@@ -46,9 +46,9 @@
 
 if ( $products->have_posts() ) : ?>
 
-	<div class="related products">
+	<section class="related products">
 
-		<h2><?php _e( 'Related Products', 'woocommerce' ); ?></h2>
+		<h2><?php echo sprintf( __( 'Other %s...', 'woocommerce' ), $product->get_categories() ); ?></h2>
```

Comment:
Probably not a huge deal @opportus as the categories are already displayed and linked in the product meta. Not in a heading tag of course, not sure how important that is though?

The only alternative I can think of would be to check how many categories the product is in and if it's less than 5 (or so) then display them in this heading, otherwise just display the generic 'related products' heading.